### PR TITLE
Proposal: Fix Registry and Serialize Return Values

### PR DESCRIPTION
**Pull request by addison-adler**
Creating this PR to ensure that roRegistry and roRegistrySection function return values which indicate success/failure of registry calls are passed through to maestro function callers and error logging.

I have not updated the unit tests yet. Wanted to verify that this code change proposal is otherwise acceptable before putting more time into it.

### CODE REVIEW

**Review by rdaci23 — CHANGES_REQUESTED** [review]

File: src/source/core/Registry.bs [hunk 1]
```diff
@@ -70,33 +88,53 @@ namespace mc
             for each key in sectionData
               value = sectionData[key]
               key = lcase(`${key}`)
-              section.write(key, formatJson(value))
+
+              if not section.write(key, formatJson(value))
+                success = false ' flip success=false if there is any failure; subsequent successes do not negate failure
+                if not success
+                  m.log.error("Failed to write registry data for registry section '" + sectionName + "', key '" + key + "'")
```
Comment (rdaci23) [inline src/source/core/Registry.bs:95]: you can use template strings such as  `Failed to write registry data for registry section  ${sectionName } , key  ${key}`

File: src/source/core/Registry.bs [hunk 2]
```diff
@@ -23,16 +23,33 @@ namespace mc
...
+      if not success
+        m.log.error("Call to roRegistrySection.write() failed.")
+        return false
+      end if
+
+      success = section.flush()
+      if not success
+        m.log.error("Call to roRegistrySection.flush() failed.")
+        return false
+      end if
```
[the remark this answers is not in the document]
Comment (addison-adler) [reply src/source/core/Registry.bs:37]: thanks, fixed!

File: src/source/core/Registry.bs (hunk 1 above)
Comment (addison-adler) [reply src/source/core/Registry.bs]: Ah, fantastic!
I wasn't sure if they were supported in the older version of bsc

Fixed all these 🙂

**Review by rdaci23 — APPROVED** [review]

**Comment by georgejecook** [comment]
It seems the CI was not running properly; there's a bunch of failing tests on this pr. Not sure if that's on the main repo or not?.

**Comment by addison-adler** [comment]
> It seems the CI was not running properly; there's a bunch of failing tests on this pr. Not sure if that's on the main repo or not?.

all good, the tests were correct! 
fixed the return types

**Comment by georgejecook** [comment]
thanks!

**Pull request merged**